### DISCUSSION TITLE
Read request-specific data file when running all requests in a collection.

### DISCUSSION THREAD

**Discussion by malgero** [thunderclient/thunder-client-support#1071] [Q&A]
When running all requests for a collection, we have the possibility to load data file to run multiple times each request.

Is there anyway to load different data files (and make a request) for each request using scripts?

I could load request-specific files but I couldn't find a way to add as interaction when running a collection.

Any idea?

**Comment by rangav** [comment]
Hi @fmalgero 

If you open Run Collection view then you can select data file for variable values.


So you are looking for to change data file for each request?

**Reply by malgero** [reply]
Hey @rangav , tks for your reply!

I need to change the data for each request so I can create request-specific json files to specify query params and expected values for testing purposes.

Is this possible?

**Reply by rangav** [reply]
You save request to separate folder and you run that folder with data file

**Reply by rangav** [reply]
Let me know your feedback, how you envision it to work with some examples

**Comment by malgero** [comment]
in folder settings it isn't possible to read data files, is it?

I know in collection it's possible and should work but my API's won't be organized as I would like. Is this the only way?

**Reply by rangav** [reply]
Currently this is the workaround to have different data files for each folder

**Reply by rangav** [reply]
If you right click on folder and select `Run All` option, then it will open runner view.

**Reply by malgero** [reply]
perfect!!! I appreciate your help!!!
